dist/function.py

```diff
@@ -189,8 +189,161 @@ def function_declaration():
+            continue
+        module = infer_module(source_file)
+
+        # Skip auto-generated files.
+        file_contents = open(source_file, 'r').read()
+        if skip_re.search(file_contents):
+            continue
+
+        # Now actually find all definitions.
+        for m in func_def_re.finditer(file_contents):
+            fn_name = m.group(1)
+            if not fn_name.startswith('__wt_') and not fn_name.startswith('__wti_'):
+                continue
+            if fn_name in fn_defs and fn_defs[fn_name].module != module:
+                print(f'{source_file}: {fn_name} is already defined in ' +
+                      f'module "{fn_defs[fn_name].module})"')
+                sys.exit(1)
+            fn_defs[fn_name] = FunctionDefinition(fn_name, module, source_file)
+
+    # Find and check all "__wt" and "__wti" function uses.
+    files = []
+    files.extend(all_c_files())
+    files.extend(all_cpp_files())
+    files.extend(all_h_files())
+    for source_file in files:
+        if source_file.startswith('../src/include/extern'):
+            continue
+        module = infer_module(source_file)
+
+        # Skip auto-generated files.
+        file_lines = open(source_file, 'r').readlines()
+        skip = False
+        for line in file_lines:
+            if skip_re.search(line):
+                skip = True
+                break
+        if skip:
+            continue
+
+        # Find all uses. Check the use of "__wti" functions.
+        in_block_comment = False
+        for line in file_lines:
+
+            # Skip block and line comments.
+            s = line.strip()
+            if s.startswith('//') or (s.startswith('/*') and s.endswith('*/')):
+                continue
+            if s == '/*':
+                in_block_comment = True
+                continue
+            if in_block_comment:
+                if s.endswith('*/'):
+                    in_block_comment = False
+                continue
+
+            # Find all function uses in each line.
+            for m in func_use_re.finditer(line):
+                fn_name = m.group(1)
+                if not fn_name.startswith('__wt_') and not fn_name.startswith('__wti_'):
+                    continue
+                # Undefined functions are either macros or test functions; just skip them.
+                if fn_name not in fn_defs:
+                    continue
+
+                # Remember whether the function was used outside of its module and of its file.
+                fn = fn_defs[fn_name]
+                if module != fn.module:
+                    fn.used_outside_of_module = True
+                if source_file != fn.source:
+                    fn.used_outside_of_file = True
+
+                # Check whether a "__wti" function is used outside of its module.
+                if fn_name.startswith('__wti_') and module != fn.module:
+                    print(f'{source_file}: {fn_name} is used outside of its module ' +
```

Comment:
This message can be printed multiple times for the same file. Either fix that or also print the line number.

Reply:
Fixed in [acfc825](https://github.com/wiredtiger/wiredtiger/pull/10667/commits/acfc825f968f01aa6feb302fe1f275bfe33ad7f3). I added a line number.